packages/expo/android/src/main/java/expo/modules/fetch/NativeRequest.kt

```diff
@@ -34,9 +36,15 @@ internal class NativeRequest(appContext: AppContext, internal val response: Nati
 
     val headers = requestInit.headers.toHeaders()
     val mediaType = headers["Content-Type"]?.toMediaTypeOrNull()
+    var reqBody = requestBody?.toRequestBody(mediaType)
+    // OkHttp requires a non-null body for POST, PATCH and PUT requests. (also for PROPPATCH and REPORT)
+    // https://github.com/expo/expo/issues/35950#issuecomment-3245173248
+    if (reqBody == null && requestInit.method in ALLOWED_METHODS_FOR_BODY) {
+      reqBody = byteArrayOf(0).toRequestBody(mediaType)
+    }
```

Comment:
```suggestion
    val reqBody = requestBody?.toRequestBody(mediaType) ?: run {
      // OkHttp requires a non-null body for POST, PATCH, and PUT requests.
      // WinterTC fetch, however, does not have this limitation.
      // Provide an empty body to make OkHttp behave like WinterTC fetch.
      // Ref: https://github.com/expo/expo/issues/35950#issuecomment-3245173248
      if (requestInit.method in METHODS_REQUIRING_BODY) {
        byteArrayOf(0).toRequestBody(mediaType)
      } else {
        null
      }
    }
```

- use elvis operator
- though okhttp also enforces body for PROPPATCH and REPORT, i would not like to mention these in the comment. we did not add these method to METHODS_REQUIRING_BODY and it's not clear whether we support these methods